game-app/game-core/src/main/java/games/strategy/engine/data/TechnologyFrontier.java

```diff
@@ -56,12 +57,28 @@ public void removeAdvance(final TechAdvance t) {
     getData().getTechTracker().clearCache();
   }
 
-  public TechAdvance getAdvanceByProperty(final String property) {
-    return techs.stream().filter(ta -> ta.getProperty().equals(property)).findAny().orElse(null);
+  /**
+   * Returns {@link TechAdvance} by search first via {@link
+   * TechnologyFrontier#getAdvanceByProperty(String)} and then by {@link
+   * TechnologyFrontier#getAdvanceByName(String)}.
+   *
+   * @param propertyOrName Search string
+   * @return {@link Optional} of {@link TechAdvance}
+   */
+  public Optional<TechAdvance> getAdvanceByPropertyOrName(final String propertyOrName) {
```

Comment:
I'm not a fan of how wordy this method name has become. But I understand why it is.. I don't have any good suggestions, but want to note it..